### PR TITLE
[NSXServiceAccount] Support backup/restore

### LINKED ISSUE #208: Support NSXT backup/restore for NSXServiceAccount
When Antrea Cluster is created after a backup point, the PI/CCP will be lost after NSXT restored to the backup point.
nsx-operator need to read PI/CCP info from CR and re-create them on NSXT.

Target release: 4.1.2

### PR DESCRIPTION
**Pull request by gran-vmv**
Fix #208

### CODE REVIEW

File: pkg/controllers/nsxserviceaccount/nsxserviceaccount_controller_test.go [hunk 1]
```diff
@@ -178,10 +178,11 @@ func TestNSXServiceAccountReconciler_Reconcile(t *testing.T) {
 						Phase: nsxvmwarecomv1alpha1.NSXServiceAccountPhaseRealized,
 					},
 				}))
-				patches = gomonkey.ApplyMethodSeq(r.Service.NSXClient, "NSXCheckVersionForNSXServiceAccount", []gomonkey.OutputCell{{
-					Values: gomonkey.Params{true},
-					Times:  1,
-				}})
+				cluster := &nsx.Cluster{}
+				patches = gomonkey.ApplyMethod(reflect.TypeOf(cluster), "GetVersion", func(_ *nsx.Cluster) (*nsx.NsxVersion, error) {
+					nsxVersion := &nsx.NsxVersion{NodeVersion: "4.0.1"}
```
[the remark this answers is not in the document]
Comment (gran-vmv) [reply pkg/controllers/nsxserviceaccount/nsxserviceaccount_controller_test.go:183]: At first, we plan to add NSXServiceAccount to 4.0.1, but actually we need 4.1.0 for VPC feature.
In this PR, the existing version compatibility is not changed.

File: pkg/nsx/services/nsxserviceaccount/cluster.go [hunk 2]
```diff
@@ -193,6 +156,94 @@ func (s *NSXServiceAccountService) CreateOrUpdateNSXServiceAccount(ctx context.C
...
+		return fmt.Errorf("CCP store is not synchronized")
+	}
+	switch err.(type) {
+	case vapierrors.NotFound:
+	default:
+		return err
+	}
+
+	// check Secret
+	log.Info("Start to restore realized", "nsxserviceaccount", types.NamespacedName{Name: obj.Name, Namespace: obj.Namespace})
```
Comment (liu4480) [inline pkg/nsx/services/nsxserviceaccount/cluster.go:183]: may I know the reason to put "nsxserviceaccount" in a single phrase other than putting it togerther with "Start to restore realized"?

**Review by liu4480 — APPROVED** [review]

File: pkg/nsx/services/nsxserviceaccount/cluster.go (hunk 2 above)
Comment (gran-vmv) [reply pkg/nsx/services/nsxserviceaccount/cluster.go:183]: This `log.Info` makes same behavior as `klog.InfoS`, thus is means `"Start to restore realized, nsxserviceaccount=%v", namespacednamme`

Comment (edwardbadboy) [reply pkg/nsx/services/nsxserviceaccount/cluster.go:193]: I feel that this line can be better written as below.

```
log.Info("Start to restore realized Kubernetes resource", "nsxserviceaccount", ...
```

**Review by edwardbadboy — APPROVED** [review]

**Review by liu4480 — APPROVED** [review]

**Pull request merged**